internal/provider/resource_definition_resource.go

```diff
@@ -254,18 +255,57 @@ func parseResourceDefinitionResponse(ctx context.Context, driverInputSchema map[
 	}
 
 	if data.DriverInputs != nil {
-		if driverInputs.SecretRefs == nil {
-			data.DriverInputs.SecretRefs = types.StringNull()
-		} else {
-			if !strings.Contains(data.DriverInputs.SecretRefs.ValueString(), `{"value":"`) {
-				b, err := json.Marshal(driverInputs.SecretRefs)
-				if err != nil {
-					diags.AddError(HUM_API_ERR, fmt.Sprintf("Failed to marshal secret_refs: %s", err.Error()))
+		diags.Append(parseResourceDefinitionSecretRefResponse(driverInputs.SecretRefs, data)...)
+	}
+	return diags
+}
+
+func parseResourceDefinitionSecretRefResponse(secretRefs *map[string]interface{}, data *DefinitionResourceModel) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if secretRefs == nil {
+		data.DriverInputs.SecretRefs = types.StringNull()
+		return diags
+	}
+
+	existingStateSecretRefs := map[string]interface{}{}
+
+	// unmarshal existing secret_refs
+	if !data.DriverInputs.SecretRefs.IsNull() {
+		existingRefs := data.DriverInputs.SecretRefs.ValueString()
+		if err := json.Unmarshal([]byte(existingRefs), &existingStateSecretRefs); err != nil {
+			diags.AddError(HUM_API_ERR, fmt.Sprintf("Failed to unmarshal existing secret_refs: %s, \"%s\"", err.Error(), existingRefs))
+			return diags
+		}
+	}
+
+	// take the API response, but set the "value" field for each secret_ref to the value of the existing state
+	newStateSecretRefs := map[string]map[string]interface{}{}
+
+	for key, secretRefI := range *secretRefs {
+		secretRef, ok := secretRefI.(map[string]interface{})
+		if !ok {
+			diags.AddError(HUM_API_ERR, fmt.Sprintf("Failed to parse secret_ref %s: %s", key, secretRefI))
+			continue
+		}
+		// value is never returned from the API, so take the value from the existing state
+		if existingRefI, ok := existingStateSecretRefs[key]; ok {
```

Comment:
Is this going to work when nested secret references? 
Nested secret refs is the reason why we implemented that ugly `value` detection...